src/main/scala/com/fulcrumgenomics/umi/VanillaUmiConsensusCaller.scala

```diff
@@ -125,16 +125,18 @@ class VanillaUmiConsensusCaller(override val readNamePrefix: String,
     val buffer = new ListBuffer[SamRecord]()
 
     // fragment
-    consensusFromSamRecords(records=fragments).map { frag => buffer += createSamRecord(read=frag, readType=Fragment) }
+    consensusFromSamRecords(records=fragments).map { frag =>
+      buffer += createSamRecord(read=frag, readType=Fragment, fragments.flatMap(_.get[String](ConsensusTags.UmiBases)))
+    }
 
     // pairs
     (consensusFromSamRecords(firstOfPair), consensusFromSamRecords(secondOfPair)) match {
       case (None, Some(r2))     => rejectRecords(secondOfPair, UmiConsensusCaller.FilterOrphan)
       case (Some(r1), None)     => rejectRecords(firstOfPair,  UmiConsensusCaller.FilterOrphan)
       case (None, None)         => rejectRecords(firstOfPair ++ secondOfPair, UmiConsensusCaller.FilterOrphan)
       case (Some(r1), Some(r2)) =>
-        buffer += createSamRecord(r1, FirstOfPair)
-        buffer += createSamRecord(r2, SecondOfPair)
+        buffer += createSamRecord(r1, FirstOfPair, firstOfPair.flatMap(_.get[String](ConsensusTags.UmiBases)))
```

Comment:
Having the UMI consensus calling in `createSamRecord` means that we're doing it twice when we only need to do it once, right?  I.e. we expect/require that UMIs on R1 and R2 are the same, so we don't _really_ need to call it twice.  How would you feel about having createSamRecord take in the consensus UMI instead to avoid this?

Reply:
Maybe if we make the method more efficient we shouldn't worry about it?